src/Entities/Entity.cpp

```diff
@@ -242,10 +242,17 @@ void cEntity::TakeDamage(eDamageType a_DamageType, cEntity * a_Attacker, int a_R
 {
 	TakeDamageInfo TDI;
 	TDI.DamageType = a_DamageType;
-	TDI.Attacker = a_Attacker;
+	if ((a_Attacker != nullptr) && a_Attacker->IsPawn())
+	{
+		TDI.Attacker = a_Attacker;
+	}
+	else
+	{
+		TDI.Attacker = nullptr;
+	}
```

Comment:
Maybe I could do it that way:
```
TDI.Attacker - The entity that caused the damage, could be a projectile or a mob or a nullptr
TDI.Shooter - If TDI.Attacker is a projectile, this is the shooter of that projectile. Otherwise this is null.
```

Then in cMonster:
```
if (TDI.shooter != nullptr) m_Target = TDI.shooter;
else m_Target = TDI.Attacker;
```

That shouldn't harm plugins.